src/libcore/time.rs

```diff
@@ -501,13 +502,81 @@ impl Mul<u32> for Duration {
     }
 }
 
+#[stable(feature = "duration_mul_div_extras", since = "1.29.0")]
+impl Mul<Duration> for u32 {
+    type Output = Duration;
+
+    fn mul(self, rhs: Duration) -> Duration {
+        rhs.checked_mul(self).expect("overflow when multiplying scalar by duration")
+    }
+}
+
+#[stable(feature = "duration_mul_div_extras", since = "1.29.0")]
+impl Mul<f64> for Duration {
+    type Output = Duration;
+
+    fn mul(self, rhs: f64) -> Duration {
+        const NPS: f64 = NANOS_PER_SEC as f64;
+        let nanos_f64 = rhs * (NPS * (self.secs as f64) + (self.nanos as f64));
+        if !nanos_f64.is_finite() {
+            panic!("got non-finite value when multiplying duration by float");
+        }
+        if nanos_f64 > (u128::MAX as f64) {
```

Comment:
The upper limit is `u64::MAX * NANOS_PER_SEC` (1.8 × 10<sup>28</sup>) which is a much smaller number than `u128::MAX` (3.4 × 10<sup>38</sup>). If you check for the former you don't need the second `secs > (u64::MAX as u128)` below.

Reply:
I was conservative about potential edge effects. So you think it will be enough to just check the following condition?
```Rust
nanos_f64 > ((u64::MAX as u128)*(NANOS_PER_SEC as u128)) as f64
```